### DISCUSSION TITLE
How to accelerate large image instance segmentation speed

### DISCUSSION THREAD

**Discussion by fanweiya** [obss/sahi#741] [Q&A] [answered]
I use sahi to do instance segmentation. my images have 11749 x 10403 ,but object have 100x100 size.it cost 20 minutes, it's  to long time.
[this my image](https://img.giao111.workers.dev/api/img/PZhz6c)

**Comment by fcakyon** [comment]
Can you be more specific?

Which function are you using from sahi?

What is your slice size and image size?

**Reply by fanweiya** [reply]
> Can you be more specific?
> 
> Which function are you using from sahi?
> 
> What is your slice size and image size?

i use sahi and mmdetection，slice size is 960×960，and my photo size is  11749 x 10403.

**Comment by fcakyon** [comment, marked as answer]
It is expected. As the number of slices increases, accuracy and inference time also increase. 

If you want inference time to get shorter, you can: 
- increase the slice_size, 
- decrease the overlap_ratio, 
- increase the model_confidence_threshold, 
- decrease the iamge_size, 
- use NMS postprocess

**Reply by fanweiya** [reply]
> It is expected. As the number of slices increases, accuracy and inference time also increase.
> 
> If you want inference time to get shorter, you can:
> 
> * increase the slice_size,
> * decrease the overlap_ratio,
> * increase the model_confidence_threshold,
> * decrease the iamge_size,
> * use NMS postprocess

I found that the time required for the slice instances segmentation was mainly due to the merging duplication mask images from different slices. Can we converting masks to segments points first, then combining segments points from different slice Implement de-duplication and merge？This can reduce memory usage and increase speed.

**Reply by fcakyon** [reply]
It is not trivial to merge segmentation points, but it is easier to merge boolean masks.

What type of algorithm do you have in mind for segmentation point merging?

**Reply by fanweiya** [reply]
I found these, but I haven't tried them yet.
https://stackoverflow.com/questions/2667748/how-do-i-combine-complex-polygons
https://github.com/fonttools/pyclipper

> It is not trivial to merge segmentation points, but it is easier to merge boolean masks.
> 
> What type of algorithm do you have in mind for segmentation point merging?
